src/ConstructiveSolidGeometry/LinePrimitives/Line.jl

```diff
@@ -0,0 +1,74 @@
+struct Line{T,TP,TL} <: AbstractLinePrimitive{T}
+    p1::TP
+    p2::TP
+    linetype::TL
+    function Line( ::Type{T},
+                   p1::Union{CartesianPoint{T}, PlanarPoint{T}},
+                   p2::Union{CartesianPoint{T}, PlanarPoint{T}},
+                   linetype::Union{Val{:inf}, Val{:ray}, Val{:seg}}) where {T}
+        new{T,typeof(p1),typeof(linetype)}(p1, p2)
+    end
+end
+
+function Line( ::Type{T},
+               p1::Union{CartesianPoint{T}, PlanarPoint{T}},
+               p2::Union{CartesianPoint{T}, PlanarPoint{T}}) where {T} #if (inf)Line there are no endpoints
+    Line(T, p1, p2, Val(:inf))
+end
+
+function Ray( ::Type{T},
+               p1::Union{CartesianPoint{T}, PlanarPoint{T}},
+               p2::Union{CartesianPoint{T}, PlanarPoint{T}}) where {T} #if Ray endpoint is p1
+    Line(T, p1, p2, Val(:ray))
+end
```

Comment:
Are these just for completeness or do we use them for the simulations?
I can only think of applications of `LineSegment` right now.

Reply:
For completeness and they do get used for intersections. For now only LineSegment is used